features/update/deduplication/feature.go

```diff
@@ -0,0 +1,90 @@
+package deduplication
+
+import (
+	"context"
+	"time"
+
+	"go.temporal.io/features/features/update/updateutil"
+	"go.temporal.io/features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	reusedUpdateID = "reused_update_id"
+	incrementCount = "incrementCount"
+	shutdownSignal = "shutdown_signal"
+	expectedCount  = 1
+)
+
+var Feature = harness.Feature{
+	Workflows: Count,
+	Execute: func(ctx context.Context, runner *harness.Runner) (client.WorkflowRun, error) {
+		if reason := updateutil.CheckServerSupportsUpdate(ctx, runner.Client); reason != "" {
+			return nil, runner.Skip(reason)
+		}
+		run, err := runner.ExecuteDefault(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		handle1, err := runner.Client.UpdateWorkflowWithOptions(
+			ctx,
+			&client.UpdateWorkflowWithOptionsRequest{
+				UpdateID:   reusedUpdateID,
+				WorkflowID: run.GetID(),
+				RunID:      run.GetRunID(),
+				UpdateName: incrementCount,
+			},
+		)
+		runner.Require.NoError(err)
+
+		handle2, err := runner.Client.UpdateWorkflowWithOptions(
+			ctx,
+			&client.UpdateWorkflowWithOptionsRequest{
+				UpdateID:   reusedUpdateID,
+				WorkflowID: run.GetID(),
+				RunID:      run.GetRunID(),
+				UpdateName: incrementCount,
+			},
+		)
+		runner.Require.NoError(err)
```

Comment:
I think you are right about idempotence.  I can see an activity doing an update, and if something else after that goes wrong, all the later retries of the activity would fail if they don't ignore the error...